Sources/DatadogSDKTesting/DDTestObserver.swift

```diff
@@ -9,120 +9,164 @@ import Foundation
 
 class DDTestObserver: NSObject, XCTestObservation {
     static let testNameRegex = try! NSRegularExpression(pattern: "([\\w]+) ([\\w]+)", options: .caseInsensitive)
-    static let supportsSkipping = NSClassFromString("XCTSkippedTestContext") != nil
     static let tracerVersion = (Bundle(for: DDTestObserver.self).infoDictionary?["CFBundleShortVersionString"] as? String) ?? "unknown"
+    
+    enum State {
+        case none
+        case module(DDTestModule)
+        case suite(DDTestSuite)
+        case test(DDTest)
+    }
 
-    var module: DDTestModule?
-    var suite: DDTestSuite?
-    var test: DDTest?
+    private(set) var state: State
 
     override init() {
         XCUIApplication.swizzleMethods
+        state = .none
         super.init()
     }
 
     func startObserving() {
         XCTestObservationCenter.shared.addTestObserver(self)
     }
+    
+    func stopObserving() {
+        XCTestObservationCenter.shared.removeTestObserver(self)
+    }
 
     func testBundleWillStart(_ testBundle: Bundle) {
-        let bundleName = testBundle.bundleURL.deletingPathExtension().lastPathComponent
+        guard case .none = state else {
```

Comment:
Maybe it would make sense to make State implement `Equatable` and use simple `==` here

Reply:
there is only one case `.none` without associated object. I think Equatable implementation will need more lines of code than 2 case unwraps. And this is internal state enum, which is used only in that observer.